src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -166,9 +166,7 @@ public abstract class HtmlDocletWriter {
 
     private final Set<String> headingIds = new HashSet<>();
 
-    protected final Map<String, String> headings = new LinkedHashMap<>();
-
-    protected ListBuilder tocBuilder;
+    protected TableOfContents tableOfContents;
```

Comment:
If you init it in the constructor for `HtmlDocletWriter`, the field can be `final`